### PR TITLE
docs: improve the plugin-develop doc

### PR DESCRIPTION
**Pull request by spacewander**
Signed-off-by: spacewander <spacewanderlzx@gmail.com>

### Pre-submission checklist:

* [ ] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [x] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

File: doc/plugin-develop.md [hunk 1]
```diff
@@ -106,33 +108,41 @@ $(INSTALL) apisix/plugins/skywalking/*.lua $(INST_LUADIR)/apisix/plugins/skywalk
 
 ## schema and check
 
-Write [Json Schema](https://json-schema.org) descriptions and check functions. similarly, take the key-auth plugin as an example to see its
+Write [Json Schema](https://json-schema.org) descriptions and check functions. similarly, take the example-plugin plugin as an example to see its
```
Comment (jbampton) [inline doc/plugin-develop.md:111]:
```suggestion
Write [Json Schema](https://json-schema.org) descriptions and check functions. Similarly, take the example-plugin plugin as an example to see its
```

File: doc/plugin-develop.md [hunk 2]
```diff
@@ -106,33 +108,41 @@ $(INSTALL) apisix/plugins/skywalking/*.lua $(INST_LUADIR)/apisix/plugins/skywalk
...
 
 ```json
- "key-auth" : {
-       "key" : "auth-one"
-  }
+"example-plugin" : {
+    i = 1,
+    s = "s",
+    t = {1}
+}
```
Comment (nic-chen) [inline doc/plugin-develop.md:119]: it should be json here，not lua table ？

Comment (spacewander) [reply doc/plugin-develop.md:119]: @nic-chen 
Solved.

File: doc/plugin-develop.md [hunk 3]
```diff
@@ -113,9 +113,9 @@ Write [Json Schema](https://json-schema.org) descriptions and check functions. S
 
 ```json
 "example-plugin" : {
-    i = 1,
-    s = "s",
-    t = {1}
+    "i": 1,
+    "s": "s",
+    "t": {1}
```
Comment (dabue) [inline doc/plugin-develop.md:118]: I thougt you  want to  use  "t":[1] .

Comment (spacewander) [reply doc/plugin-develop.md]: @dabue 
Solved.

**Review by juzhiyuan — APPROVED** [review]

**Review by nic-chen — APPROVED** [review]

**Pull request merged**